pyupgrade.py

```diff
@@ -22,26 +22,77 @@
 from tokenize_rt import tokens_to_src
 from tokenize_rt import UNIMPORTANT_WS
 
+try:
+    from typing import Text
+except ImportError:  # pragma: no cover
+    if sys.version_info < (3, 0):
+        Text = unicode  # type: ignore  # noqa
+    else:
+        Text = str  # type: ignore
+
+if False:
+    from typing import Any
+    from typing import Container
+    from typing import Dict
+    from typing import Iterable
+    from typing import Iterator
+    from typing import List
+    from typing import Match
+    from typing import Optional
+    from typing import Pattern
+    from typing import Sequence
+    from typing import Set
+    from typing import Text
+    from typing import Tuple
+    from typing import Type
+    from typing import TypeVar
+    from typing import Union
+
+    Parsed = Tuple[Text, Optional[Text], Optional[Text], Optional[Text]]
+    Fmt = Tuple[
+        Optional[Text],
+        Optional[Text],
+        Optional[Text],
+        Optional[Text],
+        Optional[Text],
+    ]
+
+    if sys.version_info < (3, 0):
+        AsyncFunctionDef = ast.stmt
+    else:
+        AsyncFunctionDef = ast.AsyncFunctionDef
+
+    T = TypeVar('T')
+
+    NameOrAttr = Union[ast.Name, ast.Attribute]
+
+try:
+    from typing import cast
+except ImportError:  # pragma: no cover
+    def cast(_, obj):
+        # type: (Type[T], Any) -> T
+        return obj
 
 _stdlib_parse_format = string.Formatter().parse
 
 
-def parse_format(s):
+def parse_format(s):  # type: (Text) -> Tuple[Parsed, ...]
     """Makes the empty string not a special case.  In the stdlib, there's
     loss of information (the type) on the empty string.
     """
-    parsed = tuple(_stdlib_parse_format(s))
+    parsed = tuple(_stdlib_parse_format(s))  # type: ignore
     if not parsed:
-        return ((s, None, None, None),)
+        return (s, None, None, None),
```

Comment:
the parens are more readable here

Reply:
i figured the type annotation would add clarity instead but i can put this back